platforms/common/uORB/uORBManager.hpp

```diff
@@ -335,6 +337,26 @@ class uORB::Manager
 	 */
 	int	orb_get_interval(int handle, unsigned *interval);
 
+	static bool orb_device_node_exists(ORB_ID orb_id, uint8_t instance);
+
+	static void *orb_add_internal_subscriber(ORB_ID orb_id, uint8_t instance, unsigned *initial_generation);
```

Comment:
I'm seeing about 1% higher usage @ 2khz on F4. When inlining these 2 (and below), it's almost back on the same level.

Reply:
The problem of inlining those two is that that would then link subscribers directly to DeviceMaster. The sole purpose of making a true link-time interface here is to get rid of that, because in protected build I don't have the DeviceMaster in user side at all.

What I could do is to add ifdefs to use inlined versions in nuttx flat build and not-inlined in protected. Would that do?

I wonder if there is something else that could be done for that. The cost is added when creating (or trying to create) subscriptions

Reply:
I will try to inline the related hot functions uORBDeviceMaster instead first and see how it looks like; it should also make smaller code than inlining the stuff into every subscriber

Reply:
Yes ifdef's could work, but let's see if you find something better. `orb_add_internal_subscriber` can be called at high frequency for topics that don't exist.

Reply:
Probably this achieves the same, inlining copy in DeviceNode and the few functions in DeviceManager. Not sure how clean solution this is though, those are somewhat large functions to be inlined. However, when they are not called from many places now, it shouldn't grow the code size considerably.

Reply:
This is good now, CPU load increase is negligible.